src/web/rustdoc.rs

```diff
@@ -483,6 +484,22 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
 
     let latest_path = format!("/crate/{}/latest{}{}", name, target_redirect, query_string);
 
+    let canonical_url = if krate.documentation_url.is_none()
+        || krate
+            .documentation_url
+            .as_ref()
+            .unwrap()
+            .starts_with("https://docs.rs/")
+    {
+        format!(
+            "https://docs.rs/{}/latest/{}",
+            name,
+            inner_path
```

Comment:
there is no guarantee that a page that exists in one version also exists in another version. 

That's why we use the `target-redirect` handler for the links in the dropdown. 

Should we put a link to the target-redirect here into the canonical url? 

Or replicate part of the logic in `target_redirect_handler` to show something more usable here? 

( since archive storage file-exist queries are relatively fast and local, but we probably should measure)

Reply:
Ah, understood, this is probably worth a comment so nobody is confused. 

at some point we could optimize this by pointing to the specific latest version where this page exists, but this would need faster file-exist checks.